### PR TITLE
Make fits_get_colnum case sensitive by default

### LINKED ISSUE #98: fits_get_colnum should be case-sensitive
It seems that fits_get_colnum is set-up to be always case insensitive.
I know a fair number of astronomical catalogs that use case-sensitive column names.
The Gaia DR2 catalog (as offered on [CDS ViZieR](http://vizier.u-strasbg.fr/viz-bin/VizieR-3?-source=I/345/gaia2)) released yesterday is one of them (e.g. the b_Teff and B_Teff columns).
It would be great to make it case sensitive by default.
thanks

### PR DESCRIPTION
**Pull request by giordano**
Add an optional keyword to select the behavior.  Fix #98.

### CODE REVIEW

File: src/libcfitsio.jl [hunk 1]
```diff
@@ -929,15 +929,15 @@ for (a,b,T) in ((:fits_get_num_cols,  "ffgncl",  :Cint),
...
+function fits_get_colnum(f::FITSFile, tmplt::String; case_sensitive::Bool=true)
     result = Ref{Cint}(0)
     status = Ref{Cint}(0)
 
     # Second argument is case-sensitivity of search: 0 = case-insensitive
     #                                                1 = case-sensitive
     ccall(("ffgcno", libcfitsio), Cint,
           (Ptr{Void}, Cint, Ptr{UInt8}, Ref{Cint}, Ref{Cint}),
-          f.ptr, 0, tmplt, result, status)
+          f.ptr, Int(case_sensitive), tmplt, result, status)
```
Comment (kbarbary) [inline src/libcfitsio.jl]: `Cint` for clarity, or maybe just leave off the conversion and ccall will convert it automatically.

File: src/table.jl [hunk 2]
```diff
@@ -444,12 +444,12 @@ function fits_read_var_col(f::FITSFile, colnum::Integer, data::Vector{String})
 end
 
 # Read a table column
-function read(hdu::ASCIITableHDU, colname::String)
+function read(hdu::ASCIITableHDU, colname::String, case_sensitive::Bool=true)
```
Comment (kbarbary) [inline src/table.jl]: `,` -> `;` ?

File: src/libcfitsio.jl (hunk 1 above)
Comment (giordano) [reply src/libcfitsio.jl]: Good point, thanks.

File: src/table.jl (hunk 2 above)
Comment (giordano) [reply src/table.jl]: Yes, of course!  Fixed, thanks

**Comment by kbarbary** [comment]
LGTM.

**Pull request merged**